### DISCUSSION TITLE
What's the purpose of this tool?

### DISCUSSION THREAD

**Discussion by geniusit** [shadcn-ui/ui#1417] [Q&A] [answered]
Im noob using ui toolkit, components libs etc and I really try to evict to use it (material-ui, chakra ui, ant design etc etc).
This is why I use tailwinds

But I don't understand, you say that you're not a component lib but it seems that almost all of your components are "just" an upon layer of radixui which is it, right ?
So why should I use shadcn-ui over radix ui over plain tailwind?

First I just tried to use the Tooltip component. I thought that if I have a button for example at very right of my screen then the tooltip will be able to show it correctly but it's not the case. The display is buggy. 

Second, in all of your examples you have a "Preview/Code" with a border that grows in white when you click on Preview / Code, it would be nice to have this kind of components but not ... 

Also the Label of shadcn seems to be just a dependency of the one of radixui where I can really see the HTML/CSS. So what is the interest to use the Label of shadcn-ui ?

I think some site like https://tailwindcomponents.com/ are better. Because there are really 100% independent as opposite to shadcn-ui which has high coupling with radixui for example.

Did I miss something ?

Thanks

**Comment by JamesSingleton** [comment, marked as answer]
I think this was kind of started as a "how do I use tailwind css with `@radix-ui/*`". I feel like it's almost an in between of just rolling everything your own and something like https://tailwindui.com/components (which uses their own `@headlessui/react` library). It is definitely convenient and sets a lot of global stuff up to use Tailwind CSS for things out of the box but is it perfect? No but not everything will be. Looking at Tailwind Components, it looks like it's using alpine under the hood to implement some of the interactive components like this [Datepicker](https://tailwindcomponents.com/component/datepicker-with-tailwindcss-and-alpinejs) component which is no different than the [Datepicker](https://ui.shadcn.com/docs/components/date-picker) using React in this component catalog (since it's not a library 😅). Ultimately the decision is up to you on if you want to roll your own or use something like this tool.
